chain/src/txhashset/txhashset.rs

```diff
@@ -148,6 +151,13 @@ impl TxHashSet {
 			header,
 		)?;
 
+		let bitmap_accumulator = {
+			let output_pmmr = ReadonlyPMMR::at(&output_pmmr_h.backend, output_pmmr_h.last_pos);
+			let mut bitmap_accumulator = BitmapAccumulator::new();
+			bitmap_accumulator.init(&mut output_pmmr.leaf_idx_iter(0))?;
+			bitmap_accumulator
+		};
+
```

Comment:
Is this code block equivalent to just lines 155-157 ?

Reply:
Yes but with some caveats. The reason it was wrapped in a block was we need `bitmap_accumulator` to be `mut` so we can call `init()` on it. but once it is initialized we don't need it to be `mut`.
So the block wraps this `mut` requirement up, hides it and returns an immutable instance.
Makes the intent of the code clearer.

We also only need the `output_pmmr` for building the `bitmap_accumulator` so no need to expose it outside the block of code. 

Its a common rust pattern to limit scope as much as possible (both for variables and `mut`).

Reply:
isn't it possible to make the constructor accept the argument of init BitmapAccumulator::new(output_pmmr.leaf_idx_iter(0)) so we don't need to play these mutable games?